cyral/resource_cyral_sidecar_listener.go

```diff
@@ -0,0 +1,422 @@
+	s.ListenerConfig.S3SettingsFromInterface(d.Get(S3SettingsKey).(*schema.Set).List())
+	s.ListenerConfig.DynamoDbSettingsFromInterface(d.Get(DynamoDbSettingsKey).(*schema.Set).List())
+	return nil
+}
+
+// resourceSidecarListener returns the schema and methods for provisioning a sidecar listener
+// Sidecar listeners API is {{baseURL}}/sidecars/:sidecarID/listeners/:listenerID
+// GET {{baseURL}}/sidecars/:sidecarID/listeners/:listenerID (Get one listener)
+// POST {{baseURL}}/sidecars/:sidecarID/listeners/ (Create a listener)
+// PUT {{baseURL}}/sidecars/:sidecarID/listeners/:listenerID (Update a listener)
+// DELETE {{baseURL}}/sidecars/:sidecarID/listeners/:listenerID (Delete a listener)
+func resourceSidecarListener() *schema.Resource {
+	return &schema.Resource{
+		Description: "Manages [sidecar listeners](https://cyral.com/docs/sidecars/sidecar-listeners).",
+		CreateContext: CreateResource(
+			ResourceOperationConfig{
+				Name:       "SidecarListenersResourceCreate",
+				HttpMethod: http.MethodPost,
+				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+					return fmt.Sprintf("https://%s/v1/sidecars/%s/listeners",
+						c.ControlPlane,
+						d.Get(SidecarIdKey).(string))
+
+				},
+				NewResourceData: func() ResourceData { return &SidecarListenerResource{} },
+				NewResponseData: func(_ *schema.ResourceData) ResponseData { return &CreateListenerAPIResponse{} },
+			}, ReadSidecarListenersConfig,
+		),
+		ReadContext: ReadResource(ReadSidecarListenersConfig),
+		UpdateContext: UpdateResource(
+			ResourceOperationConfig{
+				Name:       "SidecarListenersResourceUpdate",
+				HttpMethod: http.MethodPut,
+				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+					return fmt.Sprintf("https://%s/v1/sidecars/%s/listeners/%s",
+						c.ControlPlane,
+						d.Get(SidecarIdKey).(string),
+						d.Get(ListenerIdKey).(string))
+
+				},
+				NewResourceData: func() ResourceData { return &SidecarListenerResource{} },
+			}, ReadSidecarListenersConfig,
+		),
+		DeleteContext: DeleteResource(
+			ResourceOperationConfig{
+				Name:       "SidecarListenersResourceDelete",
+				HttpMethod: http.MethodDelete,
+				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+					return fmt.Sprintf("https://%s/v1/sidecars/%s/listeners/%s",
+						c.ControlPlane,
+						d.Get(SidecarIdKey).(string),
+						d.Get(ListenerIdKey).(string))
+				},
+			},
+		),
+		Schema: map[string]*schema.Schema{
+			ListenerIdKey: {
+				Description: "ID of the listener that will be bound to the sidecar.",
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Computed:    true,
+			},
+			SidecarIdKey: {
+				Description: "ID of the sidecar that the listener will be bound to.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			RepoTypesKey: {
+				Description: "List of repository types that the listener supports. Currently limited to one repo type, eg [\"mysql\"].",
```

Comment:
Maybe you can use https://github.com/cyralinc/terraform-provider-cyral/blob/main/cyral/utils.go#L67 like it is being used here: https://github.com/cyralinc/terraform-provider-cyral/blob/main/cyral/resource_cyral_repository.go#L118

Reply:
nice